invenio_communities/utils.py

```diff
@@ -8,10 +8,31 @@
 
 """Utilities."""
 
-from invenio_cache import current_cache
+from flask import current_app, session
+from flask_principal import Identity
+from invenio_accounts.models import Role
+from invenio_accounts.proxies import current_db_change_history
 
 from .generators import CommunityRoleNeed
-from .proxies import current_communities
+from .proxies import current_communities, current_identities_cache
+
+IDENTITY_KEY = "user-communities:"
+
+
+def add_to_community_cache(community_id, identity_id):
```

Comment:
shelve: we're adding some data fetching logic (from the cache) in `utils`, does not feel right. However it can be tackled afterwards.

Reply:
Not sure why `utils` os not right. More than fetching we are setting some values in the cache but every time we set we need to fetch them to add the new element to the list as the set func from the cache does not allow to "append". I guess we could add an `add` func in the RedisCache class that does this logic (but it would assume that the value of the key is a list/set).

Reply:
I agree, it does not feel right, because I wouldn't expect from utility function to be obliged to have application context (and current_identities_cache kinda indicates that), but I can see it was done like this previously...
In the end you are checking for duplicates in the cache, why not making the .set method of the cache class to check it ?